zuul-core/src/main/java/com/netflix/zuul/origins/OriginName.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.zuul.origins;
+
+import com.netflix.zuul.util.VipUtils;
+import java.util.Locale;
+import java.util.Objects;
+
+public final class OriginName {
+    /**
+     * The NIWS client name of the origin.  This is typically used in metrics and for configuration of NIWS
+     * {@link com.netflix.client.config.IClientConfig} objects.
+     */
+    private final String niwsClientName;
+
+    /**
+     * This should not be used in {@link #equals} or {@link #hashCode} as it is already covered by
+     * {@link #niwsClientName}.
+     */
+    private final String metricId;
```

Comment:
Do we really need this state? Don't feel too strongly though.

Reply:
I had wanted to avoid doing the alloc each time, but technically no.   I feel like it may be used alot.      If you feel strongly I wouldn't feel bad about removing it.

Reply:
If you anticipate a lot of calls, makes sense.